mlir/utils/performance/tuningRunner.py

```diff
@@ -13,60 +14,106 @@
+            print(f"""Verification failed:
+Output = {outs}
+Errors = {errs.decode('utf-8')}""")
+            return np.nan
+    except subprocess.TimeoutExpired:
+        print("Verification timed out")
+        p3.kill()
+        outs, errs = p3.communicate()
+        return np.nan
+    nanoSeconds = perfRunner.getNanoSeconds(perfRunner.BENCHMARKING_RESULT_FILE_NAME)
+    return nanoSeconds
 
 #Tune MLIR Gemm or Convolution kernels
 def tuneMLIRKernels(configs, confClass, paths: Paths, arch, rocmlir_gen_flags, debug):
+    allData = []
+    winners = {}
     for testVector in configs:
         commandLine = testVector.split(sep=' ')
         config = confClass.fromCommandLine(commandLine, arch)
         config.MLIR_N_REPEATS=1
+        print("Tuning:", testVector)
         commandLineOptions = config.generateMlirDriverCommandLine(rocmlir_gen_flags)
-        rocmlirGenCommand = paths.mlir_paths.rocmlir_gen_path + ' -ph ' + commandLineOptions + ' --emit-tuning-space'
-        args = rocmlirGenCommand.split()
-        # get tuning space for this config
-        p = subprocess.check_output(args).decode()
-        tune_df = pd.DataFrame(runMLIRWithPerfConfig(perf_config.strip(), config, paths, rocmlir_gen_flags, debug) for perf_config in p.splitlines())
-        if debug: 
-            print(tune_df)
-        print('Tuning', testVector, ':', tune_df['PerfConfig'][tune_df['TFlops'].idxmin()])
+        # Note, we don't need the -ph, this goes to the tuning driver
+        kernelGenCommand = paths.mlir_paths.rocmlir_gen_path + ' ' + commandLineOptions
+        kernelGen = subprocess.Popen(kernelGenCommand.split(), stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+        tuningLoop = subprocess.Popen([paths.mlir_paths.rocmlir_tuning_driver_path],
+            stdin=kernelGen.stdout, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        kernelGen.stdout.close()
+
+        # Tune, printing progress as we go to avoid CI timeouts
+        maxTFlops = -np.inf
+        winningConfig = "None"
+        for i, result in enumerate(tuningLoop.stdout):
+            result = result.decode('utf-8').strip()
+            if i > 0 and i % 50 == 0:
+                print(f"Tested {i} configs, best perf {maxTFlops} TFlops on perf_config {winningConfig}")
+            if debug:
+                print(result)
+            # Time is in ns
+            perfConfig, time = result.split('\t')
+            if time == "N/A":
+                nanoSeconds = np.nan
+            else:
+                nanoSeconds = float(time)
+
+            config.setPerfConfig(perfConfig)
+            entry = config.tableEntry(nanoSeconds)
+            allData.append(entry)
+            theseTFlops = entry['TFlops']
+            if not np.isnan(theseTFlops) and theseTFlops > maxTFlops:
+                maxTFlops = theseTFlops
+                winningConfig = perfConfig
+        verifyNs = verifyKernelWithPerfConfig(winningConfig, config, paths, rocmlir_gen_flags, debug)
```

Comment:
Why validating the winning config? Could we add the option of skipping the validation (which is timing out occasionally for big configurations)? 

cc: @sjw36 @jungpark-mlir

Reply:
Would it be a bad idea to add a flag like `--validation={none,CPU,heuristic}` ? In this way we can disable it when doing internal prototyping work and set it to `CPU` or `heuristic` in CI to catch bugs

Reply:
No, that's actually a pretty solid idea. I'll get on that. And I'll need to extend `rocmlir-gen` to do the right thing on `-pv_with_gpu`.

Reply:
(Or to add another flag for whether `-pv_with_gpu` strips perf_config or not)